### PR TITLE
BENCH-207: Bench 207

### PR DESCRIPTION
**Pull request by oleksiiRybak**

### CODE REVIEW

File: src/main/java/com/answerdigital/answerking/repository/CategoryRepository.java [hunk 1]
```diff
@@ -13,4 +14,6 @@ public interface CategoryRepository extends PagingAndSortingRepository<Category,
     boolean existsByNameAndIdIsNot(String name, Long id);
 
     Set<Category> findAll();
+
+    List<Category> findCategoriesByProductsId(Long productId);
```
Comment (AudriusPutys) [inline src/main/java/com/answerdigital/answerking/repository/CategoryRepository.java:18]: I think this should be singular as it finds by single product id.: 
`List<Category> findCategoriesByProductId(Long productId);`

File: src/main/java/com/answerdigital/answerking/repository/ProductRepository.java [hunk 2]
```diff
@@ -13,4 +13,6 @@ public interface ProductRepository extends PagingAndSortingRepository<Product, L
     boolean existsByNameAndIdIsNot(String name, Long id);
 
     List<Product> findAll();
+
+    List<Product> findProductsByCategoriesId(Long categoryId);
```
Comment (AudriusPutys) [inline src/main/java/com/answerdigital/answerking/repository/ProductRepository.java:17]: same as above, `findProductsByCategoryId` as it needs single category id.

File: src/main/java/com/answerdigital/answerking/controller/CategoryController.java [hunk 3]
```diff
@@ -76,6 +81,18 @@ public ResponseEntity<Category> getCategoryById(@PathVariable @NotNull final Lon
...
+    @Operation(summary = "Get all products in a category.")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "When all the products have been returned.",
+                content = { @Content(mediaType = "application/json", schema = @Schema(implementation = Category.class)) }),
+        @ApiResponse(responseCode = "404", description = "When the category with the given id does not exist.",
+                content = { @Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class)) })
+    })
+    @GetMapping("/{categoryId}/products")
+    public ResponseEntity<Collection<Product>> fetchProductsByCategory(@PathVariable @NotNull final Long categoryId) {
+        return new ResponseEntity<>(productService.findProductsByCategoryId(categoryId), HttpStatus.OK);
```
Comment (AndyFlintAnswerDigital) [inline src/main/java/com/answerdigital/answerking/controller/CategoryController.java:93]: Wouldn't it be better if this talked to the CategoryService and returned all products within the specified category?

**Review by AndyFlintAnswerDigital — CHANGES_REQUESTED** [review]
I'm always hesitant about having a controller talk to a different type of service, e.g. in this case where CategoryController is talking to ProductService. 

I feel that it'd be better if our logic was more like
1. CategoryController talks to category service
2. Category service finds category by ID
3. Category service returns the products within that category

File: src/main/java/com/answerdigital/answerking/repository/ProductRepository.java (hunk 2 above)
Comment (oleksiiRybak) [reply src/main/java/com/answerdigital/answerking/repository/ProductRepository.java]: I've done the changes.

File: src/main/java/com/answerdigital/answerking/repository/CategoryRepository.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (oleksiiRybak) [reply src/main/java/com/answerdigital/answerking/repository/CategoryRepository.java]: removed it.

File: src/main/java/com/answerdigital/answerking/controller/CategoryController.java (hunk 3 above)
Comment (oleksiiRybak) [reply src/main/java/com/answerdigital/answerking/controller/CategoryController.java]: good catch, I moved it to the CategoryService.

File: src/main/java/com/answerdigital/answerking/repository/CategoryRepository.java (hunk 1 above)
Comment (oleksiiRybak) [reply src/main/java/com/answerdigital/answerking/repository/CategoryRepository.java]: made the changes.

**Review by AudriusPutys — APPROVED** [review]

**Pull request merged**